### PR TITLE
cpu/nrf51: added I2C driver implementation

### PR DESCRIPTION
**Pull request by haukepetersen**
The implementation is not perfect but works -> the driver will be touched again anyhow with #4926 progressing...

The error handling is cut short a little bit, but for now it should do the job, tested with the `airfy-beacon` and the `bh1750fvi`.

### CODE REVIEW

**Comment by haukepetersen** [comment]
fixed issues with outdated `I2C_x` defines

File: cpu/nrf51/periph/i2c.c [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+ */
+#define INVALID_SPEED_MASK  (0x0f)
+
+/**
+ * @brief   Initialized bus locks (we have a maximum of two devices...)
+ */
+static mutex_t locks[] = {
+    MUTEX_INIT,
+    MUTEX_INIT
+};
```
Comment (PeterKietzmann) [inline cpu/nrf51/periph/i2c.c]: Is it on purpose to initialize both locks by default?

File: cpu/nrf51/periph/i2c.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    return i2c_config[bus].dev;
+}
+
+static int error(i2c_t bus)
+{
+    dev(bus)->ERRORSRC = (TWI_ERRORSRC_DNACK_Clear |
+                          TWI_ERRORSRC_ANACK_Clear |
+                          TWI_ERRORSRC_OVERRUN_Clear);
+    dev(bus)->EVENTS_ERROR = 0;
+    DEBUG("[i2c] error 0x%02x\n", (int)dev(bus)->ERRORSRC);
```
Comment (PeterKietzmann) [inline cpu/nrf51/periph/i2c.c]: Didn't you clear the `ERRORSRC` in lines 52-54?

**Comment by PeterKietzmann** [comment]
Is the change in ` pkg/u8g2/patches/0002-u8g2-add-riot-os-interface.patch ` really related to the nrf51 I2C driver or would it make more sense to make a separate PR?

**Comment by PeterKietzmann** [comment]
You should add `FEATURES_PROVIDED += periph_i2c` to the nrf board Makefiles

**Comment by PeterKietzmann** [comment]
Could you add configurations in "periph_conf.h" for the Nordic boards please?

File: cpu/nrf51/periph/i2c.c (hunk 2 above)
Comment (haukepetersen) [inline cpu/nrf51/periph/i2c.c]: true - seems like I never used that output... moved it upwards to the top of the function.

**Comment by haukepetersen** [comment]
> Is the change in pkg/u8g2/patches/0002-u8g2-add-riot-os-interface.patch really related to the nrf51 I2C driver or would it make more sense to make a separate PR?

Yes, it is related! The way the driver is implemented, you can't use `#if I2C_NUMOF` anymore with the nrf i2c driver (same for all future remodeled drivers), but Murdock wants to build that packe for all boards that offer i2c -> hence the fix is neccessary with this PR.

**Comment by haukepetersen** [comment]
- added i2c configuration for feasible boards
- fixed DEBUG message in error() function

**Comment by LudwigKnuepfer** [comment]
This should be `pkg`, not `pkt`: `pkt/u2g2: use #ifdef SPI_NUMOF instead of #if`.

**Comment by haukepetersen** [comment]
yes, thats right, fixed

**Comment by PeterKietzmann** [comment]
@haukepetersen I'd like to test this PR but I cant build e.g. `driver_srf02` for the nrf51dongle or nrf52dk

**Comment by haukepetersen** [comment]
the `nrf52dk` won't work, as this driver is for the `nrf51` only... The `nrf51dongle` won't build, because I did not configure any I2C device for that board -> makes no sense as no pins are externally available. For testing, best try the `airfy-beacon`, the `yunjia-nrf51822`, or the `microbit`. Do you have any of these available?

**Comment by haukepetersen** [comment]
rebased

**Comment by PeterKietzmann** [comment]
Dammit, you're totally right. I just  have yunjias and I hate those. Anyway -will give it a try again.

**Comment by PeterKietzmann** [comment]
@haukepetersen then why did you add configurations and `FEATURES_PROVIDED` for nrf52dk? Successfully tested driver_srf02 with yunjia. Let's see what Murdock says

**Comment by haukepetersen** [comment]
that was a mistake on my side -> fixed

**Pull request merged**